modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/CacheBasedDatasetBuilder.java

```diff
@@ -102,16 +105,24 @@ public CacheBasedDatasetBuilder(Ignite ignite, IgniteCache<K, V> upstreamCache,
             ignite,
             upstreamCache.getName(),
             filter,
+            transformersChain,
             datasetCache.getName(),
             partCtxBuilder,
             RETRIES,
             RETRY_INTERVAL
         );
 
-        return new CacheBasedDataset<>(ignite, upstreamCache, filter, datasetCache, partDataBuilder, datasetId);
+        return new CacheBasedDataset<>(ignite, upstreamCache, filter, transformersChain, datasetCache, partDataBuilder, datasetId);
     }
 
     /** {@inheritDoc} */
+    @Override public UpstreamTransformerChain<K, V> upstreamTransformersChain() {
+        return transformersChain;
```

Comment:
And one more comment about `transformerChain`. As you can see here it makes `DatasetBuilder` mutable while without this chain it was immutable.  I think we should try to save this immutability if it's possible.